### PR TITLE
docs: update comments to use consistent code formatting for boolean values

### PR DESCRIPTION
**Pull request by JeevanMahesha**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
- [x] Documentation content changes

## What is the current behavior?
The NgForOfContext class defines getter methods (first, last, even, odd) without any comments, making it less self-explanatory.

Issue Number: #57432 

## What is the new behavior?
The same class now includes comments above each getter method, clarifying what each one does, thereby improving code readability and understanding without altering functionality.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by JeanMeche** [comment]
The changes will be visible in the docs once #57622 is merged.

**Comment by JeanMeche** [comment]
To pass the linting job, could you please reword the commit and drop the scope, there is no need for one when doing doc changes.

File: packages/common/src/directives/ng_for_of.ts [hunk 1]
```diff
@@ -35,18 +35,30 @@ export class NgForOfContext<T, U extends NgIterable<T> = NgIterable<T>> {
     public count: number,
   ) {}
 
+  /*
+   * first: True when the item is the first item in the iterable.
```
Comment (JeanMeche) [inline packages/common/src/directives/ng_for_of.ts:39]:
```suggestion
   * first: `true` when the item is the first item in the iterable.
```

Comment (JeevanMahesha) [reply packages/common/src/directives/ng_for_of.ts:39]: okay @JeanMeche will update this PR.

**Comment by AndrewKushnir** [comment]
@JeevanMahesha thanks for improving Angular docs, the change looks great! Could you please merge both commits into one, keeping the "docs: update comments to use consistent code formatting for boolean values" string as a commit message?

**Comment by JeevanMahesha** [comment]
> @JeevanMahesha thanks for improving Angular docs, the change looks great! Could you please merge both commits into one, keeping the "docs: update comments to use consistent code formatting for boolean values" string as a commit message?

Done @AndrewKushnir

File: packages/common/src/directives/ng_for_of.ts [hunk 2]
```diff
@@ -35,18 +35,30 @@ export class NgForOfContext<T, U extends NgIterable<T> = NgIterable<T>> {
     public count: number,
   ) {}
 
+  /*
+   * first: `true` when the item is the first item in the iterable.
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_for_of.ts:39]:
```suggestion
   * Indicates whether this is the first item in the collection.
```

File: packages/common/src/directives/ng_for_of.ts [hunk 3]
```diff
@@ -35,18 +35,30 @@ export class NgForOfContext<T, U extends NgIterable<T> = NgIterable<T>> {
...
+   */
   get first(): boolean {
     return this.index === 0;
   }
 
+  /*
+   * last: `true` when the item is the last item in the iterable.
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_for_of.ts:46]:
```suggestion
   * Indicates whether this is the last item in the collection.
```

File: packages/common/src/directives/ng_for_of.ts [hunk 4]
```diff
@@ -35,18 +35,30 @@ export class NgForOfContext<T, U extends NgIterable<T> = NgIterable<T>> {
...
+   */
   get last(): boolean {
     return this.index === this.count - 1;
   }
 
+  /*
+   * even: `true` when the item has an even index in the iterable.
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_for_of.ts:53]:
```suggestion
   * Indicates whether an index of this item in the collection is even.
```

File: packages/common/src/directives/ng_for_of.ts [hunk 5]
```diff
@@ -35,18 +35,30 @@ export class NgForOfContext<T, U extends NgIterable<T> = NgIterable<T>> {
...
+   */
   get even(): boolean {
     return this.index % 2 === 0;
   }
 
+  /*
+   * odd: `true` when the item has an odd index in the iterable.
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_for_of.ts:60]:
```suggestion
   * Indicates whether an index of this item in the collection is odd.
```

File: packages/common/src/directives/ng_for_of.ts [hunk 6]
```diff
@@ -35,18 +35,30 @@ export class NgForOfContext<T, U extends NgIterable<T> = NgIterable<T>> {
     public count: number,
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_for_of.ts:35]: For completeness, we can also add docs for these fields:
```
    /** Reference to the current item from the collection. */
    public $implicit: T,

    /**
     * The value of the iterable expression. Useful when the expression is
     * more complex then a property access, for example when using the async pipe
     * (`userStreams | async`).
     */
    public ngForOf: U,

    /** Returns an index of the current item in the collection. */
    public index: number,

    /** Returns total amount of items in the collection. */
    public count: number,
```

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@JeevanMahesha thanks for updating the PR. The `test` CI job indicates that golden files should be updated as well. Could you please run the following command locally and include an updated file into this PR as well?

```
yarn bazel run //packages/common:common_api.accept
```

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
**Caretaker note**: docs-only change, no presubmit is required, this PR is ready for merge.

**Comment by devversion** [comment]
This PR was merged into the repository by commit 9e32cc4a2097a7d5ff0557e6fff6809e8c559b3f.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**